backend/src/main/java/com/empire/World.java

```diff
@@ -750,16 +746,16 @@ void transferGold() {
 		}
 
 		void setGothiSpellStatus() {
-			HashSet<String> passedSpells = new HashSet<>();
-			for (String g : new String[]{"Alyrja", "Rjinku", "Lyskr", "Syrjen"}) {
+			HashSet<NationData.Gothi> passedSpells = new HashSet<>();
+			for (NationData.Gothi g : NationData.Gothi.values()) {
 				int votes = 0;
 				int votesTotal = 0;
 				for (String k : kingdoms.keySet()) {
-					getNation(k).gothi.put(g, orders.getOrDefault(k, new HashMap<String, String>()).getOrDefault("gothi_" + g, getNation(k).gothi.get(g) ? "checked" : "").equals("checked"));
+					getNation(k).gothi.put(g, orders.getOrDefault(k, new HashMap<String, String>()).getOrDefault("gothi_" + g.toString().toLowerCase(), getNation(k).gothi.get(g) ? "checked" : "").equals("checked"));
 				}
 				for (Region r : regions) {
 					if (r.religion == null) continue;
-					if (r.religion.toString().toLowerCase().equals(g.toLowerCase())) {
+					if (r.religion.toString().toLowerCase().equals(g.toString().toLowerCase())) {
```

Comment:
In this case and in any other Gothi-to-Ideology comparisons, would it be better to avoid the string comparison altogether?  I can think of a few possible ways.  Give the Gothi enums a reference to the corresponding Ideology enums and/or vice versa.  Or let the Gothi just somehow be a subset of the actual Ideology enums since they are the exact same name.  The spells are closely related to the Ideology, I don't think it would hurt to make this link explicit in the code.

Reply:
Yeah, I want to do a follow-up to make that change; postponed it here.